packages/synapse-react-client/src/components/Authentication/AuthenticationMethodSelection.tsx

```diff
@@ -15,6 +15,7 @@ type AuthenticationMethodSelectionProps = {
   /* Invoked before redirecting to Google. Useful in portals where we may want to store the current URL to redirect back here. */
   onBeginOAuthSignIn?: () => void
   onSelectUsernameAndPassword: () => void
+  state?: OAuth2State
```

Comment:
This component generates the OAuth2 URL (by making a call to the backend). Pass the OAuth2 state object into the component so that it can be included in the backend.